oneview/resource_rack_manager.go

```diff
@@ -0,0 +1,239 @@
+				Computed: true,
+				Optional: true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"model": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"modified": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"part_number": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"refresh_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"remote_support_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"password": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"scopes_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"serial_number": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"initial_scope_uris": {
+				Optional: true,
+				ForceNew: true,
+				Type:     schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Set: schema.HashString,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"support_data_collection_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"support_data_collection_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"support_data_collections_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"support_state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceRackManagerCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	rm := ov.RackManager{
+		Hostname: utils.Nstring(d.Get("hostname").(string)),
+		UserName: d.Get("username").(string),
+		Password: utils.Nstring(d.Get("password").(string)),
+		Force:    d.Get("force").(bool),
+	}
+
+	if val, ok := d.GetOk("initial_scope_uris"); ok {
+		rawInitialScopeUris := val.(*schema.Set).List()
+		initialScopeUris := make([]utils.Nstring, len(rawInitialScopeUris))
+		for i, raw := range rawInitialScopeUris {
+			initialScopeUris[i] = utils.Nstring(raw.(string))
+		}
+		rm.InitialScopeUris = initialScopeUris
+	}
+	rmId, rmtError := config.ovClient.AddRackManager(rm)
+	d.SetId(rmId)
+	if rmtError != nil {
+		d.SetId("")
+		return rmtError
+	}
+
+	return resourceRackManagerRead(d, meta)
+}
+
+func resourceRackManagerRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	rm, err := config.ovClient.GetRackManagerById(d.Id())
```

Comment:
what is this id? are we considering name as id....or we are considering uri as id...what will happen in case import scenario?